tools/nymvisor/src/cli/run.rs

```
@@ -0,0 +1,82 @@
+// Copyright 2023 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::try_load_current_config;
+use crate::env::Env;
+use crate::error::NymvisorError;
+use crate::tasks::launcher::DaemonLauncher;
+use crate::tasks::upgrade_plan_watcher::start_upgrade_plan_watcher;
+use crate::tasks::upstream_poller::UpstreamPoller;
+use nym_bin_common::logging::setup_tracing_logger;
+use std::future::Future;
+use std::time::Duration;
+use tokio::runtime;
+use tokio::time::timeout;
+use tracing::{error, info};
+
+#[derive(clap::Args, Debug)]
+pub(crate) struct Args {
+    #[clap(trailing_var_arg = true)]
+    // #[clap(raw = true)]
+    daemon_args: Vec<String>,
+}
+
+pub(crate) fn execute(args: Args) -> Result<(), NymvisorError> {
+    let env = Env::try_read()?;
+    let config = try_load_current_config(&env)?;
+    if !config.nymvisor.debug.disable_logs {
+        setup_tracing_logger();
+    }
+
+    info!("starting nymvisor for {}", config.daemon.name);
+
+    // TODO: experiment with other minimal runtimes, maybe futures::executor::LocalPool
+    //
+    // well, if the creation of the runtime failed, there isn't much we could do
+    #[allow(clippy::expect_used)]
+    let rt = runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .expect("failed to create the runtime");
```

Why not return an error instead?